### PR TITLE
IRIS-4246 - fixes tests for forgot password flow

### PR DESCRIPTION
**Pull request by tortila**
Tests are passing: http://jenkins.wikia-prod:8080/view/Iris/view/preview/job/iris-auth-forgotten-password-preview/327/

Summary of changes:
- FANDOMize email suject
- delete all emails before and after each tests is executed
- users now have dedicated email accounts (https://github.com/Wikia/selenium-config/pull/53) so that it doesn't interrupt other tests
- slight refactoring

### CODE REVIEW

File: src/test/java/com/wikia/webdriver/common/core/helpers/ForgottenPasswordUserFactory.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    }
+
+    public String getPassword() {
+      return user.getPassword();
+    }
+  }
+
+  private static Credentials credentials = Configuration.getCredentials();
+
+  private static ForgottenPasswordUser user(User user, String email, String password) {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/common/core/helpers/ForgottenPasswordUserFactory.java]: MAJOR Remove this unused private "user" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)

File: src/test/java/com/wikia/webdriver/common/core/helpers/ForgottenPasswordUserFactory.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
+package com.wikia.webdriver.common.core.helpers;
+
+
+import com.wikia.webdriver.common.core.configuration.Configuration;
+import com.wikia.webdriver.common.properties.Credentials;
+import lombok.Builder;
+import lombok.Getter;
+
+public final class ForgottenPasswordUserFactory {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/common/core/helpers/ForgottenPasswordUserFactory.java]: MAJOR Add a private constructor to hide the implicit public one. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1118)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 6 issues
* ![MAJOR][MAJOR] 6 major

Watch the comments in this conversation to review them.

#### 4 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [BasePageObject.java#L131](https://github.com/Wikia/selenium-tests/blob/9fa79855cfc0550937fe563a4104b96fda4cb3ba/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java#L131): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [BasePageObject.java#L147](https://github.com/Wikia/selenium-tests/blob/9fa79855cfc0550937fe563a4104b96fda4cb3ba/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java#L147): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [BasePageObject.java#L556](https://github.com/Wikia/selenium-tests/blob/9fa79855cfc0550937fe563a4104b96fda4cb3ba/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java#L556): Remove this unused private "getTabsCount" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MAJOR][MAJOR] [BasePageObject.java#L610](https://github.com/Wikia/selenium-tests/blob/9fa79855cfc0550937fe563a4104b96fda4cb3ba/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java#L610): Remove this unused "initialTabsNumber" local variable. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1481)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: src/test/java/com/wikia/webdriver/common/core/helpers/ForgottenPasswordUserFactory.java (hunk 1 above)
Comment (tortila) [inline src/test/java/com/wikia/webdriver/common/core/helpers/ForgottenPasswordUserFactory.java]: it is used

File: src/test/java/com/wikia/webdriver/common/core/helpers/ForgottenPasswordUserFactory.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    public String getUsername() {
+      return user.getUserName();
+    }
+
+    public String getPassword() {
+      return user.getPassword();
+    }
+  }
+
+  private static Credentials credentials = Configuration.getCredentials();
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/common/core/helpers/ForgottenPasswordUserFactory.java]: MINOR Move this variable to comply with Java Code Conventions. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1213)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 6 issues
* ![MAJOR][MAJOR] 5 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

#### 4 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [BasePageObject.java#L131](https://github.com/Wikia/selenium-tests/blob/44e368e48d9a6ce281dec6138b1d39fd3de9a6eb/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java#L131): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [BasePageObject.java#L147](https://github.com/Wikia/selenium-tests/blob/44e368e48d9a6ce281dec6138b1d39fd3de9a6eb/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java#L147): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [BasePageObject.java#L556](https://github.com/Wikia/selenium-tests/blob/44e368e48d9a6ce281dec6138b1d39fd3de9a6eb/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java#L556): Remove this unused private "getTabsCount" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MAJOR][MAJOR] [BasePageObject.java#L610](https://github.com/Wikia/selenium-tests/blob/44e368e48d9a6ce281dec6138b1d39fd3de9a6eb/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java#L610): Remove this unused "initialTabsNumber" local variable. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1481)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/test/java/com/wikia/webdriver/common/core/helpers/ForgottenPasswordUserFactory.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+import lombok.Getter;
+
+public final class ForgottenPasswordUserFactory {
+
+  private ForgottenPasswordUserFactory() {
+    // no-op
+  }
+
+  @Builder
+  public static class ForgottenPasswordUser {
```
Comment (slayful) [inline src/test/java/com/wikia/webdriver/common/core/helpers/ForgottenPasswordUserFactory.java]: Consider using `@Value` and removing `@Getter` if exposing the user is OK.

File: src/test/java/com/wikia/webdriver/common/core/helpers/ForgottenPasswordUserFactory.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
+package com.wikia.webdriver.common.core.helpers;
+
+
+import com.wikia.webdriver.common.core.configuration.Configuration;
+import com.wikia.webdriver.common.properties.Credentials;
+import lombok.Builder;
+import lombok.Getter;
+
+public final class ForgottenPasswordUserFactory {
```
Comment (slayful) [inline src/test/java/com/wikia/webdriver/common/core/helpers/ForgottenPasswordUserFactory.java]: So I'd say this class is unnecessary. The ForgottenPasswordUser could be the base class which has got two factory methods.

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java [hunk 6]
```diff
@@ -579,11 +579,15 @@ private String getOtherTab(String title) {
...
+    Optional<String> newTab = driver.getWindowHandles()
+      .stream()
+      .map(handleName -> Pair.of(handleName, driver.switchTo().window(handleName).getTitle()))
+      .peek(handleTitle -> PageObjectLogging.log("Found window", String.format("Window with title %s", handleTitle), true))
+      .filter(condition)
+      .map(Pair::getKey)
+      .findFirst();
     return newTab.orElseThrow(
-        () -> new NotFoundException(String.format("Tab with title %s doesn't exist", title)));
+        () -> new NotFoundException("Tab that satisfies the condition doesn't exist"));
```
Comment (slayful) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java]: LGTM but, you can return immediately without the temporary `Optional<>` variable.

**Pull request merged**